src/rubocop.ts

```diff
@@ -7,36 +7,49 @@ import * as vscode from 'vscode';
 import { getConfig, RubocopConfig } from './configuration';
 import * as os from 'os';
 
-const tmpFileName = path.join(os.tmpdir(), '_rubocop.rb');
-
 export class RubocopAutocorrectProvider implements vscode.DocumentFormattingEditProvider {
     public provideDocumentFormattingEdits(document: vscode.TextDocument): vscode.TextEdit[] {
         const config = getConfig();
-        fs.writeFileSync(tmpFileName, document.getText());
-        const onSuccess = () => {
-            const newText = fs.readFileSync(tmpFileName).toString();
-            fs.unlink(tmpFileName);
-            return [new vscode.TextEdit(this.getFullRange(document), newText)];
-        };
         try {
-            const cmd = config.command.split(/\s+/)
-            cp.execFileSync(
-                cmd.pop(),
-                [...cmd, tmpFileName, ...getCommandArguments(tmpFileName), '--auto-correct'],
-                { cwd: getCurrentPath(document.fileName) },
-            );
+            const args = [
+                ...getCommandArguments(document.fileName),
+                "--auto-correct"
+            ];
+            const options = {
+                cwd: getCurrentPath(document.fileName),
+                input: document.getText()
+            };
+            let stdout;
+            if (config.useBundler) {
+                stdout = cp.execSync(`${config.command} ${args.join(" ")}`, options);
+            } else {
+                stdout = cp.execFileSync(config.command, args, options);
+            }
+
+            return this.onSuccess(document, stdout);
         } catch (e) {
             // if there are still some offences not fixed rubocop will return status 1
             if (e.status !== 1) {
                 vscode.window.showWarningMessage('An error occured during autocorrection');
                 console.log(e);
-                fs.unlink(tmpFileName);
                 return [];
             } else {
-                return onSuccess();
+                return this.onSuccess(document, e);
             }
         }
-        return onSuccess();
+    }
+
+    private onSuccess(document: vscode.TextDocument, stdout: Buffer) {
+        return [
+            new vscode.TextEdit(
+                this.getFullRange(document),
+                stdout
+                    .toString()
```

Comment:
I haven't worked at all with Node buffers, if there's a way to do this without doing the conversion to string first that may be better?

Reply:
By the way, the reason it needs to be like this is because:
```bash
> echo "def a; "3"; end" > bad_file.rb
> cat bad_file.rb | bundle exec rubocop --stdin bad_file.rb --auto-correct --force-exclusion --format json
{"metadata": ... {"offense_count":5,"target_file_count":1,"inspected_file_count":1}}====================
def a
  3
end
```